### DISCUSSION TITLE
created_by and updated_by fields

### DISCUSSION THREAD

**Discussion by mahirchavda** [pocketbase/pocketbase#1007] [Q&A] [answered]
I have created a urls collections with two fields: `url` and `created_by` (which is relation to users collections).
I also created a create rule as `@request.data.created_by = @request.auth.id` so that someone can not create a record on behalf of other user.

As per my understanding, I will need to add same create rules for all my collection since i don't want to allow someone to create record on behalf of other user.  Is my understanding correct? 

Also what do you think about adding `created_by` and `updated_by` fields as system fields like created and updated?

**Comment by ganigeorgiev** [comment]
> As per my understanding, I will need to add same create rules for all my collection since i don't want to allow someone to create record on behalf of other user. Is my understanding correct?

If your other collections has similar `created_by` field, then Yes - you need to define the same/similar access rule for all your collections with such field(s).

---

> Also what do you think about adding created_by and updated_by fields as system fields like created and updated?

It will be confusing to have them by default because a record could be created also by admin and we'll have to keep a separate identifier for that. Additionally with the upcoming v0.8.0 you'll be able to have multiple auth collections (aka. different type of app users each with their own set of fields) and if we want to add `created_by` and `updated_by` fields by default we also will have to add 2 more fields so that we can identify from which auth collection the user belongs to in order to handle all cases. And therefore it is better to leave the "creator" fields to the developers to decide on their own how and who can modify them (if they need the fields in the first place).

**Comment by mahirchavda** [comment, marked as answer]
Got it. Thanks for quick reply.
